client/spec/src/main/asciidoc/client.asciidoc

```diff
@@ -17,6 +17,392 @@
+                                       )
+                               )
+                       )
+               )
+       )
+
+----
+
+== Managing input objects
+
+In that case, we want to create a hero from scratch:
+[source,json]
+----
+mutation {
+  createNewHero(hero: {
+	name: "Green Lantern",
+    	costumeColor: "green",
+    	dateOfLastCheckin: "2020-03-25",
+    	equipment: null,
+    	knownEnemies: [“Wolverine, “Starlord”, Iron Man”],
+	idNumber: 456) {
+    realName
+  }
+}
+----
+We can observe that we have a multi-valued argument and we need a specific API construct for that: the InputObject.For instance:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.MUTATION)
+       .addRootField(field("createNewHero",
+               args(
+                       arg("hero", inputObject(
+                                   inputField("costumeColor", "green"),
+                                   inputField("dateOfLastCheckin", LocalDate.now()),
+                                   inputField("equipment", null),
+                                   inputField("idNumber", 456),
+                                   inputField("knownEnemies", asList("Wolverine", "Starlord", "Iron Man")),
+                                   inputField("name", "Green Lantern")
+                       ))
+               ),
+               fields(
+                       field("name"),
+                       field("nameOfKnownEnemies")
+               )
+       ));
+----
+
+== Running operations
+
+Once the operation is built (a mutation or a query), it is ready to be run against an endpoint. Let's see how to do that.
+
+There are two options depending on the way to get the data result.
+
+To get raw JSON data:
+[source,java]
+----
+// Build the request
+Request request = builder.build();
+
+// Initialize the GraphQL client
+Client client = ClientBuilder.newClient();
+
+// Get the GraphQL response
+Response response = client.target(“http://localhost:8080/graphql”).execute(request);
+
+// Check GraphQL errors if any
+if ( response.hasError ) {
+     System.err.println(“Sorry, got some GraphQL errors =” + response.getErrors());
+     // Process error and partial data here
+    }
+    
+// Get the raw JSON data
+JsonValue rawJson = response.getData();
+----
+
+To get the same data using an expected application class:
+[source,java]
+----
+// Build the request
+Request request = builder.build();
+
+// Initialize the GraphQL client
+Client client = ClientBuilder.newClient();
+
+// Get the GraphQL response
+Response<ApplicationClass> response = client.target(“http://localhost:8080/graphql”).dataType(ApplicationClass.class).execute(request);
+
+// Check GraphQL errors if any
+if ( response.hasError ) {
+     System.err.println(“Sorry, got some GraphQL errors =” + response.getErrors());
+     // Process error and partial data here
+    }
+    
+// Get the ApplicationClass
+ApplicationClass appClass = response.getData();
```

Comment:
If the `Response` type has a generic parameter, my IDE would warn me about raw usage in the example above. And the type is not required until it's used for the `getData()` call, is it? Maybe we could simply overload this method to take the type instead, so the example would simply continue:

```suggestion
// Get the ApplicationClass
ApplicationClass appClass = response.getData(ApplicationClass.class);
```